File: src/main/java/com/my/admin/app/commerce/model/document/Sale.java
package com.my.admin.app.commerce.model.document;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.io.Serializable;
import java.time.ZonedDateTime;

@Document(collection = "sales")
@Data
public class Sale implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    private Long id;

    private Order order;

    private Client client;

    private ZonedDateTime date;

    private PaymentType paymentType;

    private Cash cash;

    private Double payment;

    private Double returned;

    private Double total;
}

File: src/main/kotlin/com/bk/reactive/app/commerce/admin/my/model/document/Sale.kt
package com.bk.reactive.app.commerce.admin.my.model.document

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.time.ZonedDateTime


@Document(collection = "sales")
data class Sale(
    @Id
    val id: String?,
    val order: Order,
    val client: Client,
    val date: ZonedDateTime,
    val paymentType: PaymentType,
    val cash: Cash,
    val payment: Double,
    val returned: Double,
    val total: Double
)